libs/gi/ui-next/src/client/ArtifactCard.tsx

```diff
@@ -312,7 +312,7 @@ function Header({
             <ColorText color={mainStatLevel !== level ? 'warning' : undefined}>
               {getMainStatDisplayStr(
                 mainStatKey,
-                rarity as RarityKey,
+                rarity as ArtifactRarity,
```

Comment:
Why do we have to type assert?

Reply:
Artifact.rarity is a Int (number) from GraphQL interface, `ArtifactRarity` is 3|4|5
https://github.com/frzyc/genshin-optimizer/blob/master/apps/gi-backend/src/app/schema_gen.graphql#L33

Reply:
i thought the benefit of this was it had type safety

Reply:
getting the type for rarity to be a more strict enum is just extra work that hasnt been done yet, it will still validate it to be a int, which is better than `any`